### PR TITLE
Linting and pyproject.toml

### PR DESCRIPTION
**Pull request by marcus-oscarsson**
- Added pyproject.toml file
 - linted Python and Javacript code,
 - Some general cleanup

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Intended Audience :: End Users/Desktop",
+    "Natural Language :: English",
+    "Topic :: Scientific/Engineering",
+]
+
+[tool.poetry.dependencies]
+python = ">=3.8,<3.11"
```
Comment (oldfielj-ansto) [inline pyproject.toml:25]: @marcus-oscarsson Are we dropping Python3.7 support?

Comment (marcus-oscarsson) [reply pyproject.toml:25]: Not really, we can keep 3.7 if its needed it is indeed supported by mxcuebcore. I just thought it was time to move a head for the web part and did not really see the reason for keeping 3.7 thats all

File: mxcube3/core/components/queue.py [hunk 2]
```diff
@@ -2014,7 +2015,9 @@ def set_auto_mount_sample(self, automount, current_sample=None):
         """
         self.app.AUTO_MOUNT_SAMPLE = automount
 
-    def get_auto_mount_sample(self,):
+    def get_auto_mount_sample(
```
Comment (rhfogh) [inline mxcube3/core/components/queue.py]: Are you sure of this? Is it not the comma that should be removed? Anyway, it looks weird

**Review by rhfogh — APPROVED** [review]
Looks like pure linting plus udating teh .toml file.

GOod. The sooner it is merged teh better

File: pyproject.toml (hunk 1 above)
Comment (oldfielj-ansto) [reply pyproject.toml:25]: @marcus-oscarsson I'm happy to drop it a few months early, but we should probably update the Python version constraints for MXCuBE Core to be consistent.

**Review by oldfielj-ansto — APPROVED** [review]

**Comment by oldfielj-ansto** [comment]
LGTM.

**Review by meguiraun — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (marcus-oscarsson) [reply pyproject.toml:25]: Yes, that's true, we should just wait for that until we have double checked with the Qt side of things but I don't think it's a problem.

File: mxcube3/core/components/queue.py (hunk 2 above)
Comment (marcus-oscarsson) [reply mxcube3/core/components/queue.py]: That's a funny one, I guess black should have removed the additional comma as there were no additional arguments.  Ill fix that manually :)

**Pull request merged**